contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -135,7 +135,15 @@ contract LockDealNFT is LockDealNFTModifiers, IERC721Receiver {
 
     ///@dev don't use it if the provider is the owner or an approved caller
     function transferFromProvider(address from, uint256 poolId) public onlyApprovedProvider {
-        _approve(msg.sender, poolId);
         safeTransferFrom(from, address(this), poolId);
     }
+
+    function isApprovedForAll(
+        address owner,
+        address operator
+    ) public view override(ERC721, IERC721) returns(bool) {
+        return super.isApprovedForAll(owner, operator) || approvedProviders[operator];
+    }
```

Comment:
I'm not sure if we want to give that freedom to the providers.

Reply:
I guess the other option is to give permission to only the MultiWithdraw. But will have to make a new state variable for that.